sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedProcessorTest.java

```diff
@@ -360,27 +493,18 @@ public void before_ChangeFeedProcessorTest() {
 //            log.warn("Database delete", e);
 //        }
 //        createdDatabase = createDatabase(client, databaseId);
-        createdDatabase = getSharedCosmosDatabase(client);
     }
 
     @AfterMethod(groups = { "emulator" }, timeOut = 3 * SHUTDOWN_TIMEOUT, alwaysRun = true)
     public void afterMethod() {
-        safeDeleteCollection(createdFeedCollection);
-        safeDeleteCollection(createdLeaseCollection);
-
-        // Allow some time for the collections and the database to be deleted before exiting.
-        try {
-            Thread.sleep(500);
-        } catch (Exception e){ }
     }
 
     @AfterClass(groups = { "emulator" }, timeOut = 2 * SHUTDOWN_TIMEOUT, alwaysRun = true)
     public void afterClass() {
```

Comment:
I suspect you have client leak in your test class. this might cause CI test issues later.

the afterClass is annotated as part of emulator test group. But your partition split test is in a different test group. Due to that the afterClass may not get invoked.

Our test pattern is to not mix different test groups in the same class for simplicity.
Different test group go to different test classes, otherwise you need to be very cautious in the cleaning logic.

Reply:
@kushagraThapar FYI

Reply:
Fixed